### PR TITLE
chore: add windows ci

### LINKED ISSUE #6490: Repo: Add Windows testing to CI
### Suggestion

Windows development support (#2580) sometimes gets broken: e.g. #6488. GitHub Actions now has good prebuilt Windows images for Node testing. I think it'd be a good idea to add a Windows machine to the unit tests matrix.

### PR DESCRIPTION
**Pull request by liuxingbaoyu**
## PR Checklist

- [x] Addresses an existing open issue: fixes #6490
- [x] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
- [x] Steps in [Contributing](https://typescript-eslint.io/contributing) were taken

## Overview

Add windows ci

### CODE REVIEW

**Review by StyleShit — APPROVED** [review]
Not a maintainer, but LGTM

**Review by JoshuaKGoldberg — APPROVED** [review]
Am a maintainer and looks good to me! 😄 thanks!

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -147,9 +147,10 @@ jobs:
   unit_tests:
     name: Run Unit Tests
     needs: [build]
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
     strategy:
       matrix:
+        os: [ubuntu-latest, windows-latest]
```
Comment (JoshuaKGoldberg) [inline .github/workflows/ci.yml:153]: Actually, hmm... On second thought I'm not confident we really need Windows on both Node 16 and Node 20... my gut is that there won't be any significant changes between the two. And we already have quite a big list of CI checks.

Will wait for another @typescript-eslint/triage-team member to chime in.

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
Ok yes sorry for the fake-out @liuxingbaoyu - I don't think we want to run Windows on more than one Node version. In theory there may be changes between them, but in practice I doubt they'll come up enough to justify the extra CI usage.

For context, since my review here we've continuously gone past the allotment for allowed CI jobs in parallel on the GitHub free / open source tier. Having a full extra set of runs for a different Node version would make that situation even worse. 😞 

So, change request: let's just run `windows-latest` on the latest Node version that's also tested in `ubuntu-latest`.

**Comment by JoshuaKGoldberg** [comment]
👋 Hey @liuxingbaoyu! Just checking in, is this still something you have time for? No worries if not - I just don't want to leave it hanging.

**Comment by liuxingbaoyu** [comment]
Thank you for the reminder! I forgot about it. :)

**Comment by JoshuaKGoldberg** [comment]
Great! Whenever it's ready for re-review feel free to explicitly re-request mine through GitHub.

**Review by JoshuaKGoldberg — APPROVED** [review]
Great, thanks!

**Comment by bradzacher** [comment]
This has double the number of CI jobs that gets run for our unit tests (18 -> 36) - which has a BIG impact on our CI queue times and also PR check ready times.

I wonder if we could scope windows to just the latest nodejs version?
Checking 2 node versions on windows is nice, but ultimately isn't going to catch any extra bugs except for some super rare edge-cases.

**Comment by liuxingbaoyu** [comment]
https://github.com/typescript-eslint/typescript-eslint/pull/7695/files#diff-b803fcb7f17ed9235f1e5cb1fcd2f5d3b2838429d4368ae4c57ce4436577f03fR153-R155

We already do this.🤔

**Comment by bradzacher** [comment]
Oh sorry - I just saw a massive chunk of jobs spawn off and didn't actually verify things. We run so many I just assumed things incorrectly! My bad.

Oof - I wish there was an easier way to fuzz this so we didn't need so many jobs!

**Pull request merged**